dali/image/image.h

```diff
@@ -15,16 +15,21 @@
 #ifndef DALI_IMAGE_IMAGE_H_
 #define DALI_IMAGE_IMAGE_H_
 
+#include <array>
 #include <memory>
+#include <string>
 #include <tuple>
 #include <utility>
+
 #include "dali/common.h"
 #include "dali/error_handling.h"
 #include "dali/pipeline/operators/operator.h"
 
 
 namespace dali {
 
+static const std::array<string, 7> RegisteredImageExtensions = {".jpg", ".jpeg", ".png", ".gif",
+                                                                ".bmp", ".tif",  ".tiff"};
```

Comment:
I don't like global std::string constants - they depend on global variable initialization order and therefore cannot be used safely e.g. in registries, registeres etc.
`static const char *kRegisteredImageExtensions[] = { ".jpg", ".jpeg", ".png", ".gif", ".bmp", ".tif",  ".tiff" }` would be no worse - especially, since `string::rfind` accepts `const char *` parameters without converting to intermediate string.
C-array is as good as std::array (can be used with ranged for) and doesn't need explicit size.

Reply:
Naming: "registered" suggest that there's some form of registration and that additional extensions can be registered. Something like "supported" or "known" would better reflect the purpose of this list.

Reply:
I would recommend to sync and unify with https://github.com/NVIDIA/DALI/blob/master/dali/pipeline/operators/reader/loader/file_loader.cc#L32

Reply:
Done.